core/domain/translation_services_test.py

```diff
@@ -0,0 +1,131 @@
+
+        self.signup(self.ADMIN_EMAIL, self.ADMIN_USERNAME)
+        self.signup(self.OWNER_EMAIL, self.OWNER_USERNAME)
+
+        self.admin_id = self.get_user_id_from_email(self.ADMIN_EMAIL)
+        self.owner_id = self.get_user_id_from_email(self.OWNER_EMAIL)
+
+        self.set_admins([self.ADMIN_USERNAME])
+
+        self.exp_id = exp_fetchers.get_new_exploration_id()
+        exp = self.save_new_valid_exploration(
+            self.exp_id,
+            self.owner_id,
+            title='title',
+            category='category',
+            end_state_name='End State'
+        )
+
+        self.publish_exploration(self.owner_id, exp.id)
+        exp_services.update_exploration(
+            self.owner_id, self.exp_id, [exp_domain.ExplorationChange({
+                'cmd': exp_domain.CMD_EDIT_STATE_PROPERTY,
+                'property_name': exp_domain.STATE_PROPERTY_CONTENT,
+                'state_name': 'Introduction',
+                'new_value': {
+                    'content_id': 'content',
+                    'html': 'Please continue.'
+                }
+            })], 'Changes content.')
+
+    def test_machine_translation_with_invalid_language_code_raises_exception(
+            self):
+        with self.assertRaisesRegexp(
+            ValueError, 'Invalid target language code: invalid_language_code'
+        ):
+            translation_services.get_machine_translation_for_content_id(
+                self.exp_id,
+                'Introduction',
+                'content',
+                'invalid_language_code'
+            )
+
+    def test_get_machine_translation_checks_datastore_first(self):
+        with self.swap_to_always_raise(
+            cloud_translate_services.CLIENT,
+            'translate',
+            error=AssertionError):
+            self.assertEqual(
+                translation_services.get_machine_translation(
+                    'en', 'es', 'text to translate'),
+                'texto para traducir'
+            )
+
+    def test_get_machine_translation_with_new_translation_saves_translation(
+            self):
+        translated_text = translation_services.get_machine_translation(
+            'en', 'fr', 'hello world')
+        self.assertEqual(translated_text, 'Bonjour le monde')
+        translation = translation_fetchers.get_translation_for_text(
+            'en', 'fr', 'hello world')
+        self.assertIsNotNone(translation)
+        self.assertEqual(translation.translated_text, 'Bonjour le monde')
+
+    def test_get_machine_translation_for_content_id_with_valid_input(self):
+        translated_text = (
+            translation_services.get_machine_translation_for_content_id(
+                self.exp_id, 'Introduction', 'content', 'es')
+        )
+        self.assertEqual(translated_text, 'Por favor continua.')
+
+    def test_get_machine_translation_for_content_id_with_invalid_exploration_id_returns_none(self): # pylint: disable=line-too-long
```

Comment:
`test_get_machine_translation_is_none_if_exp_id_is_invalid`

I think you can do similar names below. Try to avoid adding pylint pragmas.

Reply:
Done. Could this cause confusion since I'm testing `get_machine_translation_for_content_id` and `get_machine_translation` is a separate method?

Reply:
Oh, good point. Yes, I think it might. Maybe test_get_content_id_translation or test_get_translation_for_content_id perhaps -- the method name is really long and it's probably a good idea to find a reasonable way to abbreviate it.

Reply:
This is no longer relevant since `get_machine_translation_for_content_id` is removed.